Classes/Controller/GetFileController.php

```diff
@@ -107,11 +117,38 @@ public function attachmentAction()
                 }
 
             case 'attachment':
-                $path = rtrim('http://' . $fedoraHost, "/") . '/fedora/objects/' . $piVars['qid'] . '/datastreams/' . $piVars['attachment'] . '/content';
+
+                $qid = $piVars['qid'];
+
+                $attachment = $piVars['attachment'];
+
+                if (is_numeric($piVars['qid'])) {
+
+                    // qid is local uid
+                    $document = $this->documentRepository->findByUid($piVars['qid']);
+
+                    $files = $document->getCurrentFileData();
+
+                    foreach ($files['download'] as $id => $file) {
```

Comment:
Why collecting all `download` elements when later only one is selected via `$fileIdArray[$attachment]`?